crates/sui-bridge-indexer/src/server.rs

```diff
@@ -0,0 +1,223 @@
+        let listener = TcpListener::bind(socket_address).await.unwrap();
+        axum::serve(listener, make_router(state)).await.unwrap();
+    })
+}
+
+pub(crate) fn make_router(state: PgBridgePersistent) -> Router {
+    Router::new()
+        .route("/", get(health_check))
+        .route(GET_TOKEN_TRANSFERS, get(get_token_transfers))
+        .with_state(state)
+}
+
+impl axum::response::IntoResponse for BridgeIndexerError {
+    // TODO: distinguish client error.
+    fn into_response(self) -> axum::response::Response {
+        match self {
+            BridgeIndexerError::InternalError(message) => (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Internal server error: {}", message),
+            )
+                .into_response(),
+            BridgeIndexerError::InvalidInput(message) => (
+                StatusCode::BAD_REQUEST,
+                format!("Invalid input: {}", message),
+            )
+                .into_response(),
+        }
+    }
+}
+
+async fn health_check() -> StatusCode {
+    StatusCode::OK
+}
+
+#[derive(Deserialize)]
+struct QueryParams {
+    page: usize,
+    per_page: usize,
+    chain_id: i32,
+    address: Option<String>,
+}
+
+#[debug_handler]
+async fn get_token_transfers(
+    params: Query<QueryParams>,
+    State(state): State<PgBridgePersistent>,
+) -> Result<Json<Vec<TokenTransferResult>>, BridgeIndexerError> {
+    let connection = &mut state.pool.get().await?;
+
+    // Destructure params
+    let QueryParams {
+        page,
+        per_page,
+        chain_id,
+        address,
+    } = params.0;
+
+    let offset = (page * per_page) as i64;
+    let limit = per_page as i64;
+
+    let sql = r#"
+        SELECT DISTINCT ON (tt.chain_id, tt.nonce)
+            tt.chain_id AS chain_id,
+            tt.nonce AS nonce,
+            tt.status AS status,
+            tt.block_height AS block_height,
+            tt.timestamp_ms AS timestamp_ms,
+            tt.txn_hash AS txn_hash,
+            tt.txn_sender AS txn_sender,
+            tt.gas_usage AS gas_usage,
+            tt.data_source AS data_source,
+            tt.is_finalized AS is_finalized,
+            ttd.destination_chain AS destination_chain,
+            ttd.recipient_address AS recipient_address,
+            ttd.token_id AS token_id,
+            ttd.amount AS amount
+        FROM token_transfer tt
+        JOIN token_transfer_data ttd
+            ON tt.chain_id = ttd.chain_id AND tt.nonce = ttd.nonce
+        WHERE tt.chain_id = $1
+          AND ($2::bytea IS NULL OR tt.txn_sender = $2::bytea)
+        ORDER BY tt.chain_id, tt.nonce,
+            CASE tt.status
+                WHEN 'Claimed' THEN 3
+                WHEN 'Approved' THEN 2
+                WHEN 'Deposited' THEN 1
+                ELSE 0
+            END DESC
+        LIMIT $3 OFFSET $4;
+    "#;
+
+    let txn_sender_bytes = match address {
+        Some(ref addr) => {
+            let addr_trimmed = addr.trim_start_matches("0x");
```

Comment:
I'm not familiar with `trim_start_matches` and don't know its behavior in some cases, can we put this into a function and write some unit tests for it?

Reply:
will do